### DISCUSSION TITLE
Protractor Page Objects Conversion to WDIO

### DISCUSSION THREAD

**Discussion by bajipogadadanda** [webdriverio/codemod#110] [Migration Issue]
Hi,
I have so many existing protractor scripts,  I am looking for the best strategy to migrate it. I am using Codemod as well.
In Protractor, I have used xpaths with the following syntax like  userName=by.xpath('//input[@id="userName"]')
but it's not working in WDIO(I am using Mocha with WDIO V8, Chromedriver and  Sauce services).
I have so many files, is there any possibility to re-use my existing page Object, or do I need to change  those to WDIO format like 
userName = $('[id='userName']')?
It will be very tough if I need to manually change all these Objects.
Pls help me what is the best way to re-use my existing PageObjects?

**Comment by christian-bromann** [comment]
> userName=by.xpath('//input[@id="userName"]')

Can be changed to `userName = $('//input[@id="userName"]')`

> It will be very tough if I need to manually change all these Objects.

This codemod is an excellent tool to help with this migration, if there is something we can add to the codemod please submit a PR so it can help others too.
